rebuildability_check.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3
+
+import argparse
+import logging
+import sys
+
+import osc.core
+import osc.conf
+
+import osclib.remote_project
+from osclib.remote_project import RemoteProject
+
+class RebuildabilityChecker(object):
+    def __init__(self, project_str):
+        self.logger = logging.getLogger('RebuildibilityChecker')
+        self.project = RemoteProject.find(project_str) # apiurl should be read from osc.conf.config['apiurl'], osclib config class looks like has different goal?
+
+    def result(self):
+        packages = self.project.get_packages(inherited = True) # package should include also project from which it comes. Recursive means include also inherited packages so reading meta and linked projects are needed
+        self.logger.debug("Packages %s" % [pkg.project_name + " -> " + pkg.name for pkg in packages])
+
+        title = "Testing rebuild of whole parent project"
+        description = "Temporary project including expanded copy of parent to verify if everything can be rebuild from scratch"
+        rebuild_project = self.project.create_subproject("Rebuild", title, description) # how to handle if it exists? Clean it and use? What if we do not have permission to create subproject?
+
+        testing_packages = [package.copy(rebuild_project.name, expand=True) for package in packages]
+        while not all([p.builds.is_finished for p in testing_packages]): # builds is object for handling building and is_finished means that all builds are finished ( or disabled )
```

Comment:
I assume this is supposed to read as pseudo code? Because you don't need to query all packages' build status just to find out that one is not finished (taking that `is_finished` is going to be an external API call).

Reply:
yeah, we decide to start with pseudo code how the API can look like and adapt it as we implement it. Maybe we can even find some project specific query to find unfinished build.